src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -100,9 +105,15 @@ public Objection createObjection(String requestId,
                 .withActionCode(actionCode)
                 .withStatus(objectionStatus)
                 .withStatusChangedOn(dateTimeSupplier.get())
+                .withId(referenceNumberGeneratorService.generateReferenceNumber())
                 .build();
 
-        return objectionRepository.save(entity);
+        try {
+            return objectionRepository.insert(entity);
```

Comment:
just wondering, should we have a pre defined number of retry attempts before exception-ing?

Reply:
I would argue that we should, but not as MVP, and would need its own story with acceptance criteria